src/pages/index.js

```diff
@@ -8,80 +8,172 @@ import { initialCards, cardData, options , cardSelector } from "../utils/constan
 import '../pages/index.css';
 import Popup from "../components/Popup";
 import { data } from "autoprefixer";
-
+import Api from "../components/Api.js"
+import PopupWithConfirmation from "../components/PopupWithConfirmation";
 
 const profileEditButton = document.querySelector("#profile-edit-button");
 const profileEditModal = document.querySelector("#profile-edit-modal");
 const profileTitle = document.querySelector("#profile-title");
 const profileDescription = document.querySelector("#profile-description");
 const profileEditForm = profileEditModal.querySelector(".modal__form");
 const profileTitleInput = document.querySelector("#edit-modal-input-title");
-const profileDescriptionInput = document.querySelector(
-  "#edit-modal-input-description"
-);
+const profileDescriptionInput = document.querySelector("#edit-modal-input-description");
 const addCardButton = document.querySelector("#profile-add-button")
 const addForm = document.querySelector("#add-form");
-const previewModal = document.querySelector("#preview");
+const profileAvatar = document.querySelector("#profile-avatar");
+const profileAvatarEdit = document.querySelector("#avatar-btn");
+const avatarForm = document.querySelector("#avatar-form");
+export const imagePreview = document.querySelector(".modal__preview");
+export const popupImage = imagePreview.querySelector(".modal__preview-image");
+export const popupImageTitle = imagePreview.querySelector(".modal__caption");
 
 const addCardPopup = new PopupWithForm('#profile-add-modal', handleAddCardSubmit );
 addCardPopup.setEventListeners();
 
 const editProfilePopup = new PopupWithForm('#profile-edit-modal', handleProfileEditSubmit);
 editProfilePopup.setEventListeners();
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const card = createCard(item);
-      section.addItem(card);
-    },
-  },
-  ".cards__container"
-);
-section.renderItems()
-
-export const imagePreview = document.querySelector(".modal__preview");
-export const popupImage = imagePreview.querySelector(".modal__preview-image");
-export const popupImageTitle = imagePreview.querySelector(".modal__caption");
-
-const userInfo = new UserInfo(profileTitle, profileDescription);
-
 const editFormValidator = new FormValidator(options, profileEditForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```